charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```
@@ -91,10 +91,26 @@ spec:
             {{- else }}
             - --prometheus-config-reloader={{ $configReloaderRegistry }}/{{ .Values.prometheusOperator.prometheusConfigReloader.image.repository }}:{{ .Values.prometheusOperator.prometheusConfigReloader.image.tag | default .Chart.AppVersion }}
             {{- end }}
+            {{- if (((.Values.prometheusOperator.prometheusConfigReloader.resources).requests).cpu) }}
             - --config-reloader-cpu-request={{ .Values.prometheusOperator.prometheusConfigReloader.resources.requests.cpu }}
+            {{- else }}
+            - --config-reloader-cpu-request=0
+            {{- end }}
```

The approach looks good. I have a suggestion for something cleaner. WDYT?

```suggestion
            - --config-reloader-cpu-request={{ (((.Values.prometheusOperator.prometheusConfigReloader.resources).requests).cpu) | default 0 }}
```

I tested locally and it worked:

```sh
→ helm template . --set prometheusOperator.prometheusConfigReloader.resources.requests.cpu=null | grep "config-reloader-cpu-request"
            - --config-reloader-cpu-request=0
```

---

@GMartinez-Sisti I agree, it looks a lot cleaner that way. Updated it...